File: GGAndroid_2.0/GGCore/GGLib/src/main/java/io/ganguo/library/ui/activity/BaseActivity.java
package io.ganguo.library.ui.activity;

import android.app.Activity;
import android.os.Bundle;

import io.ganguo.library.R;
import io.ganguo.utils.AppManager;
import io.ganguo.utils.common.ScreenAdaptors;
import io.ganguo.utils.common.UIHelper;


/**
 * Activity - 基类
 * 用于继承使用
 * <p/>
 * Created by zhihui_chen on 14-8-4.
 */
public abstract class BaseActivity extends Activity implements InitResources {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // register
        AppManager.addActivity(this);
        ScreenAdaptors.onBindScreenAdapter(this);
        // init resources
        beforeInitView();
        initView();
        initListener();
        initData();
    }

    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        // bind back action
        UIHelper.bindActionBack(this, findViewById(R.id.action_back));
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        // unregister
        AppManager.removeActivity(this);
    }
}

File: GGAndroid_4.0/core/support/support-app/src/main/kotlin/io/ganguo/app/ui/activity/BaseActivity.kt
package io.ganguo.app.ui.activity

import android.os.Bundle
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.LifecycleObserver
import io.ganguo.app.helper.activity.ActivityHelper

/**
 * FragmentActivity - 基类
 * Created by hulk on 10/20/14.
 */
@Suppress("LeakingThis")
abstract class BaseActivity : FragmentActivity(), InitResources {

    init {
        ActivityHelper.bindLifecycle(this)
    }


    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        beforeInitView()
        initView()
        initListener()
        initData()
    }


    /**
     * add lifecycleObserver to lifecycle
     * @param observer LifecycleObserver
     */
    fun addLifecycleObserver(observer: LifecycleObserver) {
        lifecycle.addObserver(observer)
    }
}
